### PR TITLE
sample: accept CLI arguments

### PR DESCRIPTION
**Pull request by pongad**
To the best of my knowledge this should work, but I haven't tried it
yet.

### CODE REVIEW

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip [hunk 1]
```diff
@@ -80,14 +80,28 @@
...
+    args = parser.parse_args()
+
+    sample_{@apiMethod.name}({@sampleFuncArgs(sample.initCode.argDefaultParams)})
 
   if __name__ == '__main__':
     main()
 @end
 
-@private sampleFuncParam(defaultParams)
+@private sampleFuncArgs(defaultParams)
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: make the function name more descriptive. I suggest something like `sampleFuncArgsFromFlags`, `flagsToSampleFuncArgs`, or `actualArgs`

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip [hunk 2]
```diff
@@ -80,14 +80,28 @@
...
+  @join param : defaultParams on ", "
+    args.{@param.identifier}
+  @end
+@end
+
+@private sampleFuncParams(defaultParams)
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: Consider having something like `formalArgs` in the name.

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip [hunk 3]
```diff
@@ -80,14 +80,28 @@
   @end
 
   def main():
+    import argparse
+
     @# FIXME: Convert argv from strings to the correct types.
-    sample_{@apiMethod.name}(*sys.argv[1:])
+    parser = argparse.ArgumentParser()
+    @join param : sample.initCode.argDefaultParams
+      parser.add_argument('--{@param.identifier}')
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: We should set the default value from the view model, preferably here where we define the flag. (We could also do it when we get the flags, manually substituting the default if no flag was provided. But the flag library should take be the one doing it....)

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip (hunk 1 above)
Comment (pongad) [reply src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: Done

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip (hunk 2 above)
Comment (pongad) [reply src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: Done

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip (hunk 3 above)
Comment (pongad) [reply src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: Done. Setting everything to empty string for now.

**Comment by pongad** [comment]
@vchudnov-g PTAL

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip [hunk 4]
```diff
@@ -31,7 +31,7 @@
 
     @end
 
-    def sample_{@apiMethod.name}({@sampleFuncParams(sample.initCode.argDefaultParams)}):
+    def sample_{@apiMethod.name}({@formalArgs(sample.initCode.argDefaultParams)}):
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: We can do this in a separate CL, but I think it would make things clearer to rename `sample.initCode.argDefaultParams`. As it's used here, these are just the sample param definitions, including but not limited to their default values. Maybe the terminal field should be just `.sampleParams`.

Like I said, we can do this separately.

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip [hunk 5]
```diff
@@ -85,23 +85,23 @@
...
 
-    sample_{@apiMethod.name}({@sampleFuncArgs(sample.initCode.argDefaultParams)})
+    sample_{@apiMethod.name}({@actualArgs(sample.initCode.argDefaultParams)})
 
   if __name__ == '__main__':
     main()
 @end
 
-@private sampleFuncArgs(defaultParams)
+@private actualArgs(defaultParams)
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: In this function and the one below, I would s/`defaultParams`/`sampleArgs`/. 

(Incidentally, we should be consistent with "args" vs "params". We can do that lazily as we work on further changes.)

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip [hunk 6]
```diff
@@ -85,23 +85,23 @@
...
   @join param : defaultParams on ", "
     args.{@param.identifier}
   @end
 @end
 
-@private sampleFuncParams(defaultParams)
+@private formalArgs(defaultParams)
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: Here, too, I suggest s/`defaultParams`/`sampleArgs`/ for clarity

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip [hunk 7]
```diff
@@ -85,23 +85,23 @@
     @# FIXME: Convert argv from strings to the correct types.
     parser = argparse.ArgumentParser()
     @join param : sample.initCode.argDefaultParams
-      parser.add_argument('--{@param.identifier}')
+      parser.add_argument('--{@param.identifier}', default='')
```
Comment (vchudnov-g) [inline src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: Update the `FIXME` above to (a) not refer to `argv` directly and (b) remind us that we will fill in the appropriate default values.

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip (hunk 5 above)
Comment (vchudnov-g) [reply src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: You changed it to `defaultArgs`, not `sampleArgs` :-)

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip (hunk 7 above)
Comment (vchudnov-g) [reply src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: (not sure whether you saw this; it got accidentally "resolved" by someone)

Also, I just noticed that this is tagged inconsistently. We are using for `@# TODO` for user-visible comments of things the developer needs to do, and `# FIXME` for things we ourselves need to fix in the templates. Could you s/`@# FIXME`/`# FIXME`/ above?

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip (hunk 5 above)
Comment (pongad) [reply src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: AHAHAHAH

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip (hunk 6 above)
Comment (pongad) [reply src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: Done

File: src/main/resources/com/google/api/codegen/py/standalone_sample.snip (hunk 7 above)
Comment (pongad) [reply src/main/resources/com/google/api/codegen/py/standalone_sample.snip]: Oh yeah you're right. Done.
